ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -184,271 +124,257 @@ class GameManager {
+        RunLoop.main.add(timer, forMode: .common)
+      } else {
+          setDefaultText(isFirst: false)
+      }
   }
 
   @objc func setDefaultText(isFirst: Bool) {
-    guard let gameParameters = gameParameters else { return }
-
-    if !isFirst {
-      countTest += 1
-    }
+      guard let gameParameters = gameParameters else { return }
 
+      hasRespondedInCurrentTrial = false
+      delegate?.setEnableButton(isEnable: false)
 
-    if gameParameters.showFixation {
-      if let image = URL(string: gameParameters.fixation), gameParameters.fixation.contains("https") {
-        delegate?.updateFixations(image: image, isStart: false, typeTime: .fixations)
+      if !isFirst {
+        endFeedbackTimestamp = CACurrentMediaTime()
+        setEndTimeViewingImage(time: endFeedbackTimestamp!, isStart: false, type: .feedback)
+        countTest += 1
       } else {
-        delegate?.updateText(text: gameParameters.fixation, color: .black, font: Constants.bigFont, isStart: false, typeTime: .fixations)
+        countTest = 0
       }
-    }
 
-    if isEndGame() { return }
-    invalidateTimers()
-    updateButtonTitle()
+      if isEndGame() {
+        handleEndOfGame()
+        return
+      }
 
-    if gameParameters.showFixation {
-      timerSetText = Timer(timeInterval: gameParameters.fixationDuration / 1000, target: self, selector: #selector(setText), userInfo: nil, repeats: false)
-      RunLoop.main.add(timerSetText!, forMode: .common)
-    } else {
-      setText()
-    }
+      updateButtonTitle()
+
+      if gameParameters.showFixation {
+          setEndTimeViewingImage(time: CACurrentMediaTime(), isStart: true, type: .fixations)
+
+          if let image = URL(string: gameParameters.fixation), gameParameters.fixation.contains("https") {
+            delegate?.updateFixations(image: image, isStart: true, typeTime: .fixations)
+          } else {
+            delegate?.updateText(text: gameParameters.fixation, color: .black, font: Constants.bigFont, isStart: true, typeTime: .fixations)
+          }
+          timerSetText = Timer(timeInterval: gameParameters.fixationDuration / 1000, target: self, selector: #selector(setText), userInfo: nil, repeats: false)
+          RunLoop.main.add(timerSetText!, forMode: .common)
+      } else {
+        setText()
+      }
   }
 
   @objc func setText() {
-    guard let gameParameters = gameParameters else { return }
+      guard let gameParameters = gameParameters else { return }
+      guard countTest < gameParameters.trials.count else {
+          handleEndOfGame()
+          return
+      }
 
-    delegate?.setEnableButton(isEnable: true)
+      setEndTimeViewingImage(time: CACurrentMediaTime(), isStart: false, type: .fixations)
 
-    text = gameParameters.trials[countTest].stimulus.en
+      startTrialTimestamp = CACurrentMediaTime()
+      setEndTimeViewingImage(time: startTrialTimestamp!, isStart: true, type: .trial)
```

Comment:
Same here, this call is not required since you changed it in line 112 to simply `break`

Reply:
You're correct. Since we modified setEndTimeViewingImage to do nothing when type is .trial and isStart is true, this call is unnecessary.
I will remove it.